### PR TITLE
Use tokio streams instead of crossbeam to fix deadlock issues

### PR DESCRIPTION
**Pull request by DaanDD**
I had issues with this package causing my whole application to hang. After some debugging I found it was because of the usage of crossbeam within an async environment. I created a PR to switch to tokio streams instead which fixes this issue. There's no need to do `try_recv` anymore in an infinite loop, instead the blocking `recv().await` call is used. I opted to use `broadcast` because it allows for `client.send` to remain non async. This meant I had to make `Record` clonable.

### CODE REVIEW

**Comment by johnmanjiro13** [comment]
@DaanDD 
Thank you for fixing bug! I haven't realized this issue.
Could you merge main branch because I fixed the test workflow?

File: Cargo.toml [hunk 1]
```diff
@@ -17,15 +17,13 @@ include = [
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-async-trait = "0.1.61"
```
Comment (johnmanjiro13) [inline Cargo.toml:20]: I'm glad to remove these dependencies 👍

**Comment by DaanDD** [comment]
@johnmanjiro13 Done!

**Comment by johnmanjiro13** [comment]
@DaanDD
I merged #25 and this pr is conflicted with main.
Please resolve it 🙏

**Comment by DaanDD** [comment]
@johnmanjiro13 Fixed the merge conflict

**Review by johnmanjiro13 — APPROVED** [review]
LGTM! Thank you.

**Comment by johnmanjiro13** [comment]
v0.4.0 was released.
https://github.com/johnmanjiro13/tokio-fluent/releases/tag/v0.4.0

**Pull request merged**